### PR TITLE
Make validate_number/3 work consistently with decimal

### PR DESCRIPTION
**Pull request by gjaldon**
Closes https://github.com/elixir-lang/ecto/issues/841

### CODE REVIEW

File: lib/ecto/changeset.ex [hunk 1]
```diff
@@ -1106,6 +1118,26 @@ defmodule Ecto.Changeset do
...
+  defp decimal_compare(result, :equal_to) do
+    Decimal.equal?(result, Decimal.new(0))
+  end
+
+  defp decimal_compare(result, :less_than_or_equal_to) do
+    decimal_compare(result, :less_than) || decimal_compare(result, :equal_to)
+  end
+
+  defp decimal_compare(result, :greater_than_or_equal_to) do
+    decimal_compare(result, :greater_than) || decimal_compare(result, :equal_to)
```
Comment (ericmj) [inline lib/ecto/changeset.ex]: Use `or` when you the operands are booleans.

**Comment by ericmj** [comment]
:shipit:

File: lib/ecto/changeset.ex (hunk 1 above)
Comment (gjaldon) [inline lib/ecto/changeset.ex]: Just made the change. Btw, why use `or` when the operands are booleans? For every other case, use `||`?

Comment (ericmj) [inline lib/ecto/changeset.ex]: `or` signals that you are working with booleans - not truish and falseish values. It is also slightly faster.

Comment (gjaldon) [inline lib/ecto/changeset.ex]: Thanks a lot for the explanation! Really useful tidbit there :)

**Pull request merged**